plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -420,6 +392,30 @@ maybeAddSubst body a s n rhs = do
             let acceptable = costIsAcceptable t && sizeIsAcceptable t
             pure acceptable
 
+-- according to the paper we could only do this when it's `OnceSafe` - the single occurrence is not
+-- inside a lambda, nor is a constructor argument?
```

Comment:
is there a reason we don't check this? According to the paper we could duplicate work.

Reply:
But `effectSafe` says to inline a `nonstrict` binding unconditionally?

Reply:
I guess in PIR, inlining a lazy binding cannot duplicate work. e.g., suppose `x` is lazy, in `let x = <big> in ...x...x...`, `<big>` is evaluated twice whether `x` is inlined or not (unlike GHC). So the "Safe" part of "OnceSafe" does not apply.